File: src/test/java/com/example/retailstore/strategy/AffiliateDiscountStrategyTest.java
package com.example.retailstore.strategy;

import com.example.retailstore.model.Bill;
import org.junit.jupiter.api.Test;

import static com.example.retailstore.model.UserType.*;
import static org.junit.jupiter.api.Assertions.assertEquals;

class AffiliateDiscountStrategyTest {
    @Test
    void shouldCalculateNetPayableAmountWithNoGroceries() {
        Bill bill = new Bill(1000, false, AFFILIATE);

        AffiliateDiscountStrategy strategy = new AffiliateDiscountStrategy();

        double actual = strategy.calculateNetPayableAmount(bill);

        assertEquals(850, actual);
    }
}

File: src/test/java/com/example/retailstore/strategy/AffiliateDiscountStrategyTest.kt
package com.example.retailstore.strategy

import com.example.retailstore.model.Bill
import com.example.retailstore.model.UserType
import org.junit.jupiter.api.Assertions.assertEquals
import org.junit.jupiter.api.Test


internal class AffiliateDiscountStrategyTest {
    @Test
    fun shouldCalculateNetPayableAmountWithNoGroceries() {
        val bill = Bill(1000.0, false, UserType.AFFILIATE)
        val strategy = AffiliateDiscountStrategy()

        val actual = strategy.calculateNetPayableAmount(bill)

        assertEquals(850.0, actual)
    }
}